### PR TITLE
Update task that looks for existing packages

### PR DESCRIPTION
**Pull request by jeremy-audet**
A task in `tasks/main.yml` is described as checking whether yaourt is
installed. This description is incorrect. The task actually checks to
see whether a user-supplied list of packages is installed. Fix this
error.

Let this task always run in normal mode, even when check mode is
enabled. It's safe for this task to do so, as this task's purpose is to
gather information, not to change the state of the target system. This
make this role behave better when check mode is enabled, as the variable
created by this task (`archlinux_aur_check_result`) will now be created.
Plus, the ansible-lint tool is happier with the result.

### CODE REVIEW

**Comment by jeremy-audet** [comment]
Tested against two hosts. On both, the packages to be installed (cower and pacaur) were already present. If the packages to be installed aren't yet present, I think check mode still blows up.

Without this change, the following scenario occurs:

```console
$ ansible-playbook site.yml --check
[snip!]
TASK [Gathering Facts] ***************************************************************************************************************
ok: [pine.ichimonji10.name]      
ok: [beech.ichimonji10.name]     
                                                                                                                                      
TASK [jtyr.archlinux_aur : Test distribution] ****************************************************************************************
ok: [beech.ichimonji10.name] => {
    "changed": false,                                                                                                                 
    "msg": "All assertions passed"                                 
}                                
ok: [pine.ichimonji10.name] => {                                                                                                      
    "changed": false,            
    "msg": "All assertions passed"                                                                                                    
}                                                                                                                                     

TASK [jtyr.archlinux_aur : Install build dependencies] *******************************************************************************
ok: [pine.ichimonji10.name]                                                                                                           
ok: [beech.ichimonji10.name]                                                                                                          

TASK [jtyr.archlinux_aur : Check if yaourt is installed] *****************************************************************************
skipping: [pine.ichimonji10.name] => (item=cower) 
skipping: [beech.ichimonji10.name] => (item=cower) 
skipping: [pine.ichimonji10.name] => (item=pacaur) 
skipping: [beech.ichimonji10.name] => (item=pacaur) 

TASK [jtyr.archlinux_aur : include] **************************************************************************************************
included: /home/ichimonji10/code/impedimenta/ansible/galaxy/jtyr.archlinux_aur/tasks/build.yaml for beech.ichimonji10.name, pine.ichimonji10.name
included: /home/ichimonji10/code/impedimenta/ansible/galaxy/jtyr.archlinux_aur/tasks/build.yaml for beech.ichimonji10.name, pine.ichimonji10.name

TASK [jtyr.archlinux_aur : Download the package snapshot] ****************************************************************************
fatal: [beech.ichimonji10.name]: FAILED! => {"failed": true, "msg": "The conditional check 'archlinux_aur_force or item.rc != 0\n' failed. The error was: error while evaluating conditional (archlinux_aur_force or item.rc != 0\n): 'dict object' has no attribute 'rc'\n\nThe error appears to have been in '/home/ichimonji10/code/impedimenta/ansible/galaxy/jtyr.archlinux_aur/tasks/build.yaml': line 3, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Download the package snapshot\n  ^ here\n"}
fatal: [pine.ichimonji10.name]: FAILED! => {"failed": true, "msg": "The conditional check 'archlinux_aur_force or item.rc != 0\n' failed. The error was: error while evaluating conditional (archlinux_aur_force or item.rc != 0\n): 'dict object' has no attribute 'rc'\n\nThe error appears to have been in '/home/ichimonji10/code/impedimenta/ansible/galaxy/jtyr.archlinux_aur/tasks/build.yaml': line 3, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Download the package snapshot\n  ^ here\n"}
        to retry, use: --limit @/home/ichimonji10/code/impedimenta/ansible/site.retry

PLAY RECAP ***************************************************************************************************************************
beech.ichimonji10.name     : ok=5    changed=0    unreachable=0    failed=1   
pine.ichimonji10.name      : ok=5    changed=0    unreachable=0    failed=1
```

With this change, check mode behaves better:

```console
$ ansible-playbook site.yml --check
[snip!]
TASK [Gathering Facts] ***************************************************************************************************************
ok: [pine.ichimonji10.name]      
ok: [beech.ichimonji10.name]     

TASK [jtyr.archlinux_aur : Test distribution] ****************************************************************************************
ok: [beech.ichimonji10.name] => {                                  
    "changed": false,            
    "msg": "All assertions passed"                                 
}                                
ok: [pine.ichimonji10.name] => { 
    "changed": false,            
    "msg": "All assertions passed"                                 
}                                

TASK [jtyr.archlinux_aur : Install build dependencies] *******************************************************************************
ok: [pine.ichimonji10.name]      
ok: [beech.ichimonji10.name]     

TASK [jtyr.archlinux_aur : Check if packages to be installed are already installed] **************************************************
ok: [pine.ichimonji10.name] => (item=cower)                        
ok: [beech.ichimonji10.name] => (item=cower)                       
ok: [beech.ichimonji10.name] => (item=pacaur)                      
ok: [pine.ichimonji10.name] => (item=pacaur)                       

TASK [jtyr.archlinux_aur : include] **************************************************************************************************
skipping: [beech.ichimonji10.name] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'cmd': [u'pacman', u'-Q', u'cower'], u'end': u'2017-08-22 11:01:53.297555', '_ansible_no_log': False, u'stdout': u'cower 17-2', '_ansible_item_result': True, u'changed': False, 'item': u'cower', u'delta': u'0:00:00.009493', u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'chdir': None, u'_raw_params': u'pacman -Q cower', u'removes': None, u'warn': True, u'_uses_shell': False}}, 'stdout_lines': [u'cower 17-2'], 'failed_when_result': False, u'start': u'2017-08-22 11:01:53.288062', 'failed': False}) 
skipping: [beech.ichimonji10.name] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'cmd': [u'pacman', u'-Q', u'pacaur'], u'end': u'2017-08-22 11:01:53.589125', '_ansible_no_log': False, u'stdout': u'pacaur 4.7.10-1', '_ansible_item_result': True, u'changed': False, 'item': u'pacaur', u'delta': u'0:00:00.009807', u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'chdir': None, u'_raw_params': u'pacman -Q pacaur', u'removes': None, u'warn': True, u'_uses_shell': False}}, 'stdout_lines': [u'pacaur 4.7.10-1'], 'failed_when_result': False, u'start': u'2017-08-22 11:01:53.579318', 'failed': False}) 
skipping: [pine.ichimonji10.name] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'cmd': [u'pacman', u'-Q', u'cower'], u'end': u'2017-08-22 11:01:53.283877', '_ansible_no_log': False, u'stdout': u'cower 17-2', '_ansible_item_result': True, u'changed': False, 'item': u'cower', u'delta': u'0:00:00.004048', u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'chdir': None, u'_raw_params': u'pacman -Q cower', u'removes': None, u'warn': True, u'_uses_shell': False}}, 'stdout_lines': [u'cower 17-2'], 'failed_when_result': False, u'start': u'2017-08-22 11:01:53.279829', 'failed': False}) 
skipping: [pine.ichimonji10.name] => (item={'_ansible_parsed': True, 'stderr_lines': [], u'cmd': [u'pacman', u'-Q', u'pacaur'], u'end': u'2017-08-22 11:01:53.612040', '_ansible_no_log': False, u'stdout': u'pacaur 4.7.10-1', '_ansible_item_result': True, u'changed': False, 'item': u'pacaur', u'delta': u'0:00:00.003920', u'stderr': u'', u'rc': 0, u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'chdir': None, u'_raw_params': u'pacman -Q pacaur', u'removes': None, u'warn': True, u'_uses_shell': False}}, 'stdout_lines': [u'pacaur 4.7.10-1'], 'failed_when_result': False, u'start': u'2017-08-22 11:01:53.608120', 'failed': False})
[snip!]
```

File: tasks/main.yaml [hunk 1]
```diff
@@ -12,11 +12,12 @@
   tags:
     - archlinux_aur_deps
 
-- name: Check if yaourt is installed
-  shell: pacman -Q {{ item }}
+- name: Check if packages to be installed are already installed
+  command: pacman -Q {{ item }}
```
Comment (jtyr) [inline tasks/main.yaml]: Please change it back to `shell`. The rest looks ok.

**Review by jtyr — CHANGES_REQUESTED** [review]

**Comment by jeremy-audet** [comment]
Done.

Why prefer `shell` over `command`?

**Comment by jtyr** [comment]
Thanks for your contribution.

**Pull request merged**